### PR TITLE
Modal component(with Modal hook) for global use

### PR DESCRIPTION
**Pull request by BrettEastman**
- add useModal hook
- add Modal component with basic CSS style for general use for anyone who needs it

### CODE REVIEW

File: client/source/Modal.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  position: absolute;
+  cursor: pointer;
+  top: 10px;
+  right: 20px;
+  border: 0;
+  font-size: 1.3rem;
+`;
+
+export default function Modal({ children, isOpen, onClose }) {
+  return (
```
Comment (AedonGrunGott) [inline client/source/Modal.jsx]: We need to block scrolling of the main page while the modal is open. ~~I suggest adding here...~~

IGNORE BELOW!
```
  if (isOpen) {
    document.body.classList.add('stop-scrolling');
  } else {
    document.body.classList.remove('stop-scrolling');
  }
```

This adds a custom class I snuck into the global styles.

**Review by aimeekang — CHANGES_REQUESTED** [review]
Thanks for setting this up! Did you want to put this in a group folder for organization per August's suggestion? If so, we should probably let the group know, as I know my path to App and other global files may need to change reference if we add a folder.

File: client/source/useModal.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  const [showModal, setShowModal] = useState(false);
+
+  return {
+    isOpen: showModal,
+    onOpen: () => setShowModal(true),
+    onClose: () => setShowModal(false)
+  };
+};
+
+export default useModal;
```
Comment (AedonGrunGott) [inline client/source/useModal.js:13]: Nice, breaking the logic and render apart. 👍

**Review by AedonGrunGott — CHANGES_REQUESTED** [review]
Just need to add the scroll lock, and make sure it works for you.

**Comment by AedonGrunGott** [comment]
Oh yeah, and @aimeekang pointed out it shouldn't live in the root of client.

File: client/source/Modal.jsx (hunk 1 above)
Comment (AedonGrunGott) [reply client/source/Modal.jsx]: Actually, put it in the logic section. That's better!

File: client/source/useModal.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { useState } from 'react';
+
+const useModal = function useModal() {
+  const [showModal, setShowModal] = useState(false);
+
+  return {
+    isOpen: showModal,
+    onOpen: () => setShowModal(true),
```
Comment (AedonGrunGott) [inline client/source/useModal.js]: onOpen: () => { document.body.classList.add('stop-scrolling'); setShowModal(true); },
    onClose: () => { document.body.classList.remove('stop-scrolling'); setShowModal(false); }

File: client/source/Modal.jsx (hunk 1 above)
Comment (BrettEastman) [reply client/source/Modal.jsx]: Good call, that works on my end now!

File: client/source/useModal.js (hunk 2 above)
Comment (BrettEastman) [reply client/source/useModal.js:13]: Thanks!

File: client/source/useModal.js (hunk 3 above)
Comment (BrettEastman) [reply client/source/useModal.js]: Okay, I put that in there too

Comment (AedonGrunGott) [reply client/source/useModal.js]: Sorry I changed my mind, confusing I know! Should probably only live here.

**Review by AedonGrunGott — APPROVED** [review]
allfather-approves-approved

**Review by aimeekang — APPROVED** [review]

**Pull request merged**